pages/folder/[id]/index.tsx

```diff
@@ -0,0 +1,127 @@
+  const [keyword, setKeyword] = useState("");
+  const [isOpenModal, setIsOpenModal] = useState(false);
+  const [filteredLinks, setFilteredLinks] = useState<LinkData[]>([]);
+
+  const { data: foldersData, isLoading } = useFolder("/folders");
+  const { data: linksData } = useSWR(`/folders/${router.query.id}/links`);
+
+  const folders: Folder[] = foldersData ?? [];
+  const links: LinkData[] = linksData ?? [];
+
+  const folderNames = folders.map((folder) => folder.name);
+
+  const handleAddLink = (link: string) => {
+    setAddLinkValue(link);
+  };
+
+  const handleCloseModal = () => setIsOpenModal(false);
+
+  const handleOnChangeInput = (e: ChangeEvent<HTMLInputElement>) => {
+    setKeyword(e.target.value);
+    const searchedLinks = checkMatchedAllLinks(e.target.value, links);
+    setFilteredLinks(searchedLinks.length !== 0 ? searchedLinks : []);
+  };
+
+  const handleDeletekeyword = () => {
+    setKeyword("");
+    setFilteredLinks(links);
+  };
+
+  const handleSelectedFolder = (category: string) => {
+    setSelected(category);
+    const selectedFolderId = folders.find((folder: Folder) => folder.name === category)?.id ?? "";
+    updateFolderName(category);
+    router.push(`/folder/${selectedFolderId}`);
+  };
+
+  useEffect(() => {
+    if (linksData) {
+      setFilteredLinks(linksData);
+    }
+    setSelected(selectedFolderName);
+  }, [linksData, selectedFolderName]);
+
+  return (
+    <>
+      {isOpenModal && (
+        <ModalPortal>
+          <ModalContainer onClose={handleCloseModal}>
+            <AddFolders onClose={handleCloseModal} />
+          </ModalContainer>
+        </ModalPortal>
+      )}
+
+      <Layout footerRef={ref}>
+        <FolderHero
+          onChangeAddLink={handleAddLink}
+          addLinkValue={addLinkValue}
+          isFixedInput={isIntersecting}
+        />
+        <S.Contents>
+          <Searchbar
+            keyword={keyword}
+            handleOnChangeInput={handleOnChangeInput}
+            handleDelete={handleDeletekeyword}
+          />
+          {keyword && (
+            <S.SearchInfo>
+              <S.SearchKeyword>{keyword}</S.SearchKeyword>
+              으로 검색한 결과입니다.
+            </S.SearchInfo>
+          )}
+          <S.MenuContainer>
+            <Categories
+              categories={["전체", ...folderNames]}
+              selected={selected}
+              onClick={handleSelectedFolder}
+            />
+            <S.AddFolderBtn onClick={() => setIsOpenModal(true)}>
+              <span>폴더 추가</span>
+              <S.IconAdd />
+            </S.AddFolderBtn>
+          </S.MenuContainer>
+          {isLoading && <Loading />}
+          <S.MenuContainer>
+            <S.SubTitle>{selected}</S.SubTitle>
+            {selected !== "전체" && <Options selected={selected} />}
+          </S.MenuContainer>
+          {!isLoading && links.length === 0 ? (
+            <S.Blank>저장된 링크가 없습니다</S.Blank>
+          ) : (
+            <CardList links={filteredLinks} />
```

Comment:
`filteredLinks`를 state로 두지 않는 것이 조금 더 괜찮을 것 같습니다. 

```suggestion
            <CardList links={checkMatchedAllLinks(keyword, links)} />
```